### ISSUE TITLE
training get stuck without crashing

### ISSUE THREAD

**Issue by m3rcury6** [princeton-vl/CornerNet#71]
I'm able to get all the images loaded (although I'm using them as symlinks, shouldn't be an issue), but then the training just stays at 0% for hours. i'm not sure what's missing. any idea where it's getting stuck? 

```
# python3 train.py CornerNet --threads 1
loading all datasets...
using 1 threads
loading from cache file: ./cache/coco_trainval2014.pkl
loading annotations into memory...
Done (t=12.92s)
creating index...
index created!
loading from cache file: ./cache/coco_minival2014.pkl
loading annotations into memory...
Done (t=1.44s)
creating index...
index created!
system config...
{'batch_size': 49,
 'cache_dir': './cache',
 'chunk_sizes': [4, 5, 5, 5, 5, 5, 5, 5, 5, 5],
 'config_dir': './config',
 'data_dir': './data',
 'data_rng': <mtrand.RandomState object at 0x7f4be41cd2d0>,
 'dataset': 'MSCOCO',
 'decay_rate': 10,
 'display': 5,
 'learning_rate': 0.0005,
 'max_iter': 500000,
 'nnet_rng': <mtrand.RandomState object at 0x7f4be41cd318>,
 'opt_algo': 'adam',
 'prefetch_size': 5,
 'pretrain': None,
 'result_dir': './results',
 'sampling_function': 'kp_detection',
 'snapshot': 5000,
 'snapshot_name': 'CornerNet',
 'stepsize': 450000,
 'test_split': 'testdev',
 'train_split': 'trainval',
 'val_iter': 100,
 'val_split': 'minival',
 'weight_decay': False,
 'weight_decay_rate': 1e-05,
 'weight_decay_type': 'l2'}
db config...
{'ae_threshold': 0.5,
 'border': 128,
 'categories': 80,
 'data_aug': True,
 'gaussian_bump': True,
 'gaussian_iou': 0.7,
 'gaussian_radius': -1,
 'input_size': [511, 511],
 'lighting': True,
 'max_per_image': 100,
 'merge_bbox': False,
 'nms_algorithm': 'exp_soft_nms',
 'nms_kernel': 3,
 'nms_threshold': 0.5,
 'output_sizes': [[128, 128]],
 'rand_color': True,
 'rand_crop': True,
 'rand_pushes': False,
 'rand_samples': False,
 'rand_scale_max': 1.4,
 'rand_scale_min': 0.6,
 'rand_scale_step': 0.1,
 'rand_scales': array([0.6, 0.7, 0.8, 0.9, 1. , 1.1, 1.2, 1.3]),
 'special_crop': False,
 'test_scales': [1],
 'top_k': 100,
 'weight_exp': 8}
len of db: 118287
start prefetching data...
building model...
module_file: models.CornerNet
start prefetching data...
shuffling indices...
./data/coco/images/trainval2014/COCO_train2014_000000354668.jpg
shuffling indices...
./data/coco/images/minival2014/COCO_val2014_000000092091.jpg
./data/coco/images/trainval2014/COCO_train2014_000000126907.jpg
./data/coco/images/trainval2014/COCO_val2014_000000333150.jpg
./data/coco/images/minival2014/COCO_val2014_000000580197.jpg
./data/coco/images/trainval2014/COCO_val2014_000000527219.jpg
./data/coco/images/minival2014/COCO_val2014_000000153527.jpg
./data/coco/images/trainval2014/COCO_train2014_000000141160.jpg
./data/coco/images/trainval2014/COCO_train2014_000000248833.jpg
./data/coco/images/minival2014/COCO_val2014_000000471869.jpg
./data/coco/images/minival2014/COCO_val2014_000000555005.jpg
./data/coco/images/trainval2014/COCO_train2014_000000077973.jpg
./data/coco/images/minival2014/COCO_val2014_000000465822.jpg
./data/coco/images/trainval2014/COCO_train2014_000000344806.jpg
./data/coco/images/trainval2014/COCO_train2014_000000310868.jpg
./data/coco/images/minival2014/COCO_val2014_000000311295.jpg
./data/coco/images/trainval2014/COCO_val2014_000000238065.jpg
./data/coco/images/minival2014/COCO_val2014_000000232563.jpg
./data/coco/images/trainval2014/COCO_val2014_000000528030.jpg
./data/coco/images/minival2014/COCO_val2014_000000229849.jpg
./data/coco/images/trainval2014/COCO_val2014_000000310572.jpg
./data/coco/images/minival2014/COCO_val2014_000000367818.jpg
./data/coco/images/trainval2014/COCO_train2014_000000444173.jpg
./data/coco/images/minival2014/COCO_val2014_000000468577.jpg
./data/coco/images/trainval2014/COCO_val2014_000000074656.jpg
./data/coco/images/minival2014/COCO_val2014_000000545594.jpg
./data/coco/images/trainval2014/COCO_val2014_000000313948.jpg
./data/coco/images/minival2014/COCO_val2014_000000561465.jpg
./data/coco/images/trainval2014/COCO_train2014_000000257416.jpg
./data/coco/images/minival2014/COCO_val2014_000000306139.jpg
./data/coco/images/trainval2014/COCO_train2014_000000245810.jpg
./data/coco/images/minival2014/COCO_val2014_000000548246.jpg
./data/coco/images/trainval2014/COCO_val2014_000000141158.jpg
./data/coco/images/minival2014/COCO_val2014_000000113403.jpg
./data/coco/images/trainval2014/COCO_train2014_000000272297.jpg
./data/coco/images/trainval2014/COCO_val2014_000000527375.jpg
./data/coco/images/minival2014/COCO_val2014_000000509258.jpg
./data/coco/images/minival2014/COCO_val2014_000000005503.jpg
./data/coco/images/trainval2014/COCO_val2014_000000043613.jpg
./data/coco/images/minival2014/COCO_val2014_000000118367.jpg
./data/coco/images/trainval2014/COCO_train2014_000000185884.jpg
./data/coco/images/trainval2014/COCO_train2014_000000517521.jpg
./data/coco/images/minival2014/COCO_val2014_000000428280.jpg
./data/coco/images/trainval2014/COCO_val2014_000000015690.jpg
./data/coco/images/minival2014/COCO_val2014_000000267537.jpg
./data/coco/images/trainval2014/COCO_train2014_000000023927.jpg
./data/coco/images/minival2014/COCO_val2014_000000109916.jpg
./data/coco/images/trainval2014/COCO_train2014_000000330054.jpg
./data/coco/images/minival2014/COCO_val2014_000000180135.jpg
./data/coco/images/trainval2014/COCO_val2014_000000368220.jpg
./data/coco/images/minival2014/COCO_val2014_000000005600.jpg
./data/coco/images/minival2014/COCO_val2014_000000122745.jpg
./data/coco/images/trainval2014/COCO_train2014_000000388312.jpg
./data/coco/images/trainval2014/COCO_val2014_000000262758.jpg
./data/coco/images/minival2014/COCO_val2014_000000248616.jpg
./data/coco/images/trainval2014/COCO_train2014_000000309260.jpg
./data/coco/images/minival2014/COCO_val2014_000000058655.jpg
./data/coco/images/trainval2014/COCO_train2014_000000317995.jpg
./data/coco/images/minival2014/COCO_val2014_000000338901.jpg
./data/coco/images/trainval2014/COCO_train2014_000000455353.jpg
./data/coco/images/trainval2014/COCO_val2014_000000338592.jpg
./data/coco/images/minival2014/COCO_val2014_000000126226.jpg
./data/coco/images/trainval2014/COCO_val2014_000000562856.jpg
./data/coco/images/trainval2014/COCO_train2014_000000482525.jpg
./data/coco/images/minival2014/COCO_val2014_000000098839.jpg
./data/coco/images/trainval2014/COCO_val2014_000000123810.jpg
./data/coco/images/minival2014/COCO_val2014_000000312552.jpg
./data/coco/images/trainval2014/COCO_train2014_000000027793.jpg
./data/coco/images/trainval2014/COCO_train2014_000000252283.jpg
./data/coco/images/minival2014/COCO_val2014_000000364322.jpg
./data/coco/images/trainval2014/COCO_val2014_000000428116.jpg
./data/coco/images/minival2014/COCO_val2014_000000022589.jpg
./data/coco/images/trainval2014/COCO_val2014_000000329011.jpg
./data/coco/images/minival2014/COCO_val2014_000000495146.jpg
./data/coco/images/trainval2014/COCO_train2014_000000469605.jpg
./data/coco/images/minival2014/COCO_val2014_000000464476.jpg
./data/coco/images/trainval2014/COCO_train2014_000000559021.jpg
./data/coco/images/minival2014/COCO_val2014_000000280918.jpg
./data/coco/images/trainval2014/COCO_train2014_000000388508.jpg
./data/coco/images/minival2014/COCO_val2014_000000084362.jpg
./data/coco/images/trainval2014/COCO_train2014_000000291501.jpg
./data/coco/images/minival2014/COCO_val2014_000000573094.jpg
./data/coco/images/trainval2014/COCO_train2014_000000005684.jpg
./data/coco/images/trainval2014/COCO_val2014_000000249246.jpg
./data/coco/images/minival2014/COCO_val2014_000000457884.jpg
./data/coco/images/trainval2014/COCO_train2014_000000393268.jpg
./data/coco/images/minival2014/COCO_val2014_000000424721.jpg
./data/coco/images/trainval2014/COCO_train2014_000000277616.jpg
./data/coco/images/minival2014/COCO_val2014_000000533536.jpg
./data/coco/images/trainval2014/COCO_train2014_000000222276.jpg
./data/coco/images/minival2014/COCO_val2014_000000300913.jpg
./data/coco/images/minival2014/COCO_val2014_000000068628.jpg
./data/coco/images/minival2014/COCO_val2014_000000022479.jpg
./data/coco/images/minival2014/COCO_val2014_000000460494.jpg
./data/coco/images/minival2014/COCO_val2014_000000202001.jpg
./data/coco/images/minival2014/COCO_val2014_000000320632.jpg
./data/coco/images/minival2014/COCO_val2014_000000351331.jpg
./data/coco/images/minival2014/COCO_val2014_000000525600.jpg
total parameters: 201035212
setting learning rate to: 0.0005
training start...
  0%|                                                | 0/500000 [00:00<?, ?it/s]

**Comment by heilaw** [comment]
How many GPUs were you using? In the provided configuration file, we assume there are 10 GPUs on the system. If you are running the scripts with less than 10 GPUs, the training may get stuck. If you are using 10 GPUs, it may be related to [this](https://github.com/pytorch/pytorch/issues/1637#issuecomment-338268158).

**Comment by lililiiiiiiiiii** [comment]
hello!have you solve the problem???

**Comment by kjgonzalez** [comment]
hello all, i never got around to fixing this. i was trying out multiple networks at the time, and decided to go with a different one. i leave this to @heilaw to either close or respond to. thank you.

**Comment by ghost** [comment]
@heilaw My system only 1 GPU,How about set code? Thank!

**Comment by lizhe960118** [comment]
@m3rcury6 how to solve the problem?
